PSReadLine/Render.cs

```diff
@@ -992,7 +992,23 @@ private void MoveCursor(int newCursor)
                 return;
             }
 
+            if (point.Y == _console.BufferHeight)
+            {
+                // Adjust the initial cursor position and the to-be-set cursor position
+                // after scrolling up the buffer.
+                _initialY -= 1;
+                point.Y -= 1;
+                _console.SetCursorPosition(point.X, point.Y);
+
+                // The cursor top exceeds the buffer height, so we need to
+                // scroll up the buffer by 1 line.
+                _console.Write("\n");
+            }
+            else
+            {
             _console.SetCursorPosition(point.X, point.Y);
+            }
```

Comment:
How about change to the following?

```c#
            if (point.Y == _console.BufferHeight)
            {
                // Adjust the initial cursor position and the to-be-set cursor position
                // after scrolling up the buffer.
                _initialY -= 1;
                point.Y -= 1;

                // The cursor top exceeds the buffer height, so we need to
                // scroll up the buffer by 1 line.
                _console.Write("\n");
            }

            _console.SetCursorPosition(point.X, point.Y);
```

Reply:
The cursor position may not yet be on the last line, so `console.write("\n")` may not produce the desired result of scrolling the buffer, so it needs to insure the cursor is on the bottom line first.   This can be demonstrated similar as in the reported issue, using a single line that wraps two console lines, pressing <kbd>END</kbd> from the beginning.

However, I definitely missed a format issue there in the commit.

Reply:
That make sense, thanks for the clarification! Can you please add a comment explain why `SetCursorPosition` should happen before `_console.Write("\n")`?

Reply:
I've revised all the comments in this section, as they were no longer in order from where they were copied.  I think this will make things clearer.